service/src/java/org/apache/hive/service/servlet/HS2HAHealthChecker.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hive.service.servlet;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class HS2HAHealthChecker extends HS2LeadershipStatus {
+
+  @Override
+  public void doDelete(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
```

Comment:
why do we need this?

Reply:
HS2LeadershipStatus has 2 functions: 
1. Leader status (GET route)
2. Failover initiation (DELETE route)

Health check endpoint is supposed to allow only getting the leadership status. This is what is used for health checking.

The new health check endpoint is going to be not secured by LDAP and it meant for health checking only.
This is needed to disable the ability to initiate failover via health check route.

Reply:
it's a bit confusing that we extend `HS2HAHealthChecker` from `HS2LeadershipStatus`. Should it be the other way around?